src/auto-evo/simulation/PopulationSimulation.cs

```diff
@@ -134,139 +131,78 @@ private static void RunSimulationStep(SimulationConfiguration parameters, List<S
-
-            var sunlightInPatch = biome.Compounds[Sunlight].Dissolved * Constants.AUTO_EVO_SUNLIGHT_ENERGY_AMOUNT;
-
-            var hydrogenSulfideInPatch = biome.Compounds[HydrogenSulfide].Density
-                * biome.Compounds[HydrogenSulfide].Amount * Constants.AUTO_EVO_COMPOUND_ENERGY_AMOUNT;
-
-            var glucoseInPatch = (biome.Compounds[Glucose].Density
-                * biome.Compounds[Glucose].Amount
-                + patch.GetTotalChunkCompoundAmount(Glucose)) * Constants.AUTO_EVO_COMPOUND_ENERGY_AMOUNT;
-
-            var ironInPatch = patch.GetTotalChunkCompoundAmount(Iron) * Constants.AUTO_EVO_COMPOUND_ENERGY_AMOUNT;
-
-            // Begin of new auto-evo prototype algorithm
-
-            var speciesEnergies = new Dictionary<MicrobeSpecies, float>(species.Count);
-
-            var totalPhotosynthesisScore = 0.0f;
-            var totalChemosynthesisScore = 0.0f;
-            var totalChemolithoautotrophyScore = 0.0f;
-            var totalGlucoseScore = 0.0f;
-
-            var totalPredationScore = 0.0f;
-
-            // Calculate the total scores of each type in the current patch
+            var energyBySpecies = new Dictionary<MicrobeSpecies, float>();
             foreach (var currentSpecies in species)
             {
-                totalPhotosynthesisScore += GetCompoundUseScore(currentSpecies, Sunlight);
-                totalChemosynthesisScore += GetCompoundUseScore(currentSpecies, HydrogenSulfide);
-                totalChemolithoautotrophyScore += GetCompoundUseScore(currentSpecies, Iron);
-                totalGlucoseScore += GetCompoundUseScore(currentSpecies, Glucose);
-                totalPredationScore += GetPredationScore(currentSpecies);
+                energyBySpecies[currentSpecies] = 0.0f;
             }
 
-            // Avoid division by 0
-            totalPhotosynthesisScore = Math.Max(MathUtils.EPSILON, totalPhotosynthesisScore);
-            totalChemosynthesisScore = Math.Max(MathUtils.EPSILON, totalChemosynthesisScore);
-            totalChemolithoautotrophyScore = Math.Max(MathUtils.EPSILON, totalChemolithoautotrophyScore);
-            totalGlucoseScore = Math.Max(MathUtils.EPSILON, totalGlucoseScore);
-            totalPredationScore = Math.Max(MathUtils.EPSILON, totalPredationScore);
-
-            // Calculate the share of environmental energy captured by each species
-            var energyAvailableForPredation = 0.0f;
-
-            foreach (var currentSpecies in species)
+            var niches = new List<IFoodSource>
             {
-                var currentSpeciesEnergy = 0.0f;
-
-                currentSpeciesEnergy += sunlightInPatch
-                    * GetCompoundUseScore(currentSpecies, Sunlight) / totalPhotosynthesisScore;
-
-                currentSpeciesEnergy += hydrogenSulfideInPatch
-                    * GetCompoundUseScore(currentSpecies, HydrogenSulfide) / totalChemosynthesisScore;
+                new PhotosyntheticFoodSource(patch),
```

Comment:
I think better names for these would be like environmental food source, and limited compound food source. Because chemosynthesis is performed by a few specific organelles, so the naming in the code is confusing for me at least because terms that are used during gameplay are used differently here.

Reply:
See, both of those names sound too generic to me. All food sources are from the "environment", and prey species have "limited compounds"

Reply:
Wouldn't the names be *better*, then? As they apply to future compound cases as well.
If you want slap in environmental in the limited food source, and maybe name the other "patch level constant food source".

Reply:
But they're all constant by the patch. The difference is that photosynthesis calculates total energy slightly differently (https://github.com/Revolutionary-Games/Thrive/pull/2395/files#diff-5ef2a2af7e7495a66a2d7e97f255f818f642906b88ce01b1935197cdd8d975f8R15), and microbes don't need to move around to find the food, while chemosythesis requires that a microbe actively hunts compound clouds.

Reply:
I meant in the sense that constant = environmental. So meaning that those are available uniformly everywhere in the patch, in contrast to the limited compounds that could in theory be depleted (and that could be simulated) and also the microbes need to move (or rely on currents) to happen upon that resource.

Reply:
I've now gone halfway: I renamed chemosynthetic to compound, but I kept photosynthetic the same, since right now that class isn't generic enough to be used for any other uniform sources of energy. When those energy sources enter the game, we can always refactor and rename.

Reply:
I'm not super confident on that as I've seen too many PRs that go the minimal architectural, hacky route, and not doing proper refactoring, so I fear it won't get done later if it isn't done now...

Reply:
Naming a class to pretend to do something is very different then actually refactoring code. If there was another environmental energy source, we could have a conversation about altering the class to be able to do that to, but I can't write code to functionality that doesn't exist yet. Maybe sunlight and this other thing will be moved to some dictionary and the class can have a param like compounds. Maybe the two will be very different and I need to pass in a whole lambda. Maybe they will end up being so different we both agree there should be a new class. Maybe there won't be any other environmental food sources until this whole algorithm gets refactored away.

If there was a regression in functionality with this, I could see your case, as if this was some weird code that would need to be torn up to be extended. But right now this is just a class name.

Reply:
Maybe I should read through this code properly, but to me it seems simple, you'll have stuff like:
```
new UniformEnvironmentalFoodSource("sunlight", Constants.SUNLIGHT_ENERGY_CONTENT);
new UniformEnvironmentalFoodSource("heat", Constants.HEAT_ENERGY_CONTENT);
```
Something like that, so the difference is just what compound amount is looked up to give energy to the species that can use it. But if you absolutely hate that, I can read through this PR properly before commenting again.

Reply:
Having read the code properly now, it seems to be that I'm right. The only non-general part of PhotosyntheticFoodSource is that currently it's not possible to configure the compound it looks for and the modifier value for it's amount. So I don't see why the class needs to be so specifically named as it could with very small change actually work in general for any dissolved compound used in processes as input to get energy.

Reply:
Would you like it to take in a string and throw an exception on any string other than "sunlight"?

Reply:
No... Like in that example code, I'd say it should either take in the string or a Compound instance *and* the resource multiplier. That way the class will work with any Compound. And the only limit is that it should be used on Compounds where it makes sense. No need to make a general class on purpose act in a non-general way.

Reply:
Ok, that makes a lot of sense. I failed to see the second param.